### PR TITLE
[DISCUSSION] Add babel plugin

### PR DESCRIPTION
**Pull request by doug-wade**
Hey gang, this one isn't quite ready, but I'm working on a fix for #247 .  Instead of running the gulp plugin, I'm running the transform in the babel webpack loader, and wanted to get some early feedback on the general approach.

### CODE REVIEW

File: packages/babel-plugin-react-server/.travis.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+language: node_js
+node_js:
+  - "0.12"
+  - "iojs"
+  - v4
+  - v5
```
Comment (gigabo) [inline packages/babel-plugin-react-server/.travis.yml]: Does travis do anything with this?  I thought it just looked in the repo root...

File: packages/babel-plugin-react-server/package.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
+  "version": "0.0.0",
+  "description": "React Server transpilation",
+  "repository": "Redfin/babel-plugin-react-server",
+  "author": "Doug Wade <doug.wade@redfin.com> <doug.wade@redfin.com>",
```
Comment (gigabo) [inline packages/babel-plugin-react-server/package.json]: Email address got doubled up.

File: packages/babel-plugin-react-server/package.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
+  "version": "0.0.0",
+  "description": "React Server transpilation",
+  "repository": "Redfin/babel-plugin-react-server",
```
Comment (gigabo) [inline packages/babel-plugin-react-server/package.json]: I think we've generally used lower-case `redfin/` for other packages.

File: packages/babel-plugin-react-server/test/fixtures/configurable-token/.babelrc [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "plugins": [
+    ["../../src" { "token": "BAR" }]
```
Comment (gigabo) [inline packages/babel-plugin-react-server/test/fixtures/configurable-token/.babelrc]: Missing a comma?

File: packages/babel-preset-react-server/index.js [hunk 5]
```diff
@@ -1,19 +1,10 @@
...
-    require('babel-plugin-transform-es2015-computed-properties'),
-    require('babel-plugin-transform-es2015-constants'),
-    require('babel-plugin-transform-es2015-destructuring'),
-    require('babel-plugin-transform-es2015-modules-commonjs'),
-    require('babel-plugin-transform-es2015-parameters'),
-    require('babel-plugin-transform-es2015-shorthand-properties'),
-    require('babel-plugin-transform-es2015-spread'),
-    require('babel-plugin-transform-es2015-template-literals'),
-    require('babel-plugin-transform-object-rest-spread'),
+    require('babel-plugin-react-server'),
```
Comment (gigabo) [inline packages/babel-preset-react-server/index.js]: Nice!

**Comment by gigabo** [comment]
> wanted to get some early feedback on the general approach.

:metal:

**Comment by doug-wade** [comment]
This has api changes; rebasing onto `next`

**Pull request closed without merging**